Containerfile.task

```diff
@@ -4,7 +4,15 @@
 # in-cluseter with the tasks focused on building remotely. Therefore, we need to maintain a buildah image
 # for our tasks that has more than _just_ buildah in it. We also need to add the required functionality
 # for the remote builds.
+
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21 AS builder
+
+WORKDIR /app
+RUN git clone https://github.com/konflux-ci/dockerfile-json.git .
+RUN go build
```

Comment:
Can we add https://github.com/konflux-ci/dockerfile-json.git as a submodule instead, making it consistent with the way `buildah` is handled here?

Reply:
Also, every time dockerfile-json gets an update this repo will get a renovate PR, thus prompting a rebuild. Otherwise, if we wanted to get an updated dockerfile-json into this image we would have to merge some arbitrary unrelated change to trigger the pipeline

Reply:
This approach does make more sense. It's neat, did not have a chance to use it before. Thanks !